runtime/bcutil/cfreader.c

```diff
@@ -2130,11 +2130,16 @@ checkAttributes(J9CfrClassFile* classfile, J9CfrAttribute** attributes, U_32 att
 */
 
 static I_32 
-checkClassVersion(J9CfrClassFile* classfile, U_8* segment, U_32 flags)
+checkClassVersion(J9CfrClassFile* classfile, U_8* segment, U_32 vmVersionShifted)
 {
 	U_32 errorCode = J9NLS_CFR_ERR_MAJOR_VERSION__ID;
 	U_32 offset = 6;
-	U_16 max_allowed_version = flags >> BCT_MajorClassFileVersionMaskShift;
+	U_16 max_allowed_version = vmVersionShifted >> BCT_MajorClassFileVersionMaskShift;
+
+	/* protect the callers with default values */
+	if (0 == max_allowed_version) {
+		max_allowed_version = BCT_Java9MajorVersionShifted >> BCT_MajorClassFileVersionMaskShift;
```

Comment:
This needs to be fixed not to default to anything. We need to consistently know which version of Java the VM is currently running, since it supports multiple versions.

Reply:
Removed this check and fixed tests / cfdumper.